wip-First look to the guide (#25)
netgroup-polito/crownlabs-site: content/resources/golang/golang.md

+++
fragment = "content"
#disabled = true
date = "2020-12-12"
weight = 100
# title = "Simple guide: Golang language"
+++

## Introduction

> Crownlabs backend is written in Golang (a.k.a. Go), an open source, procedural programming language created by Google. To simplify it can be compared with C and C++ but without some of their most tricky issues (e.g. memory management). This guide will do a general excursus on this very raising language introducing its most relevant features, giving a roadmap to follow to learn it.

Reviewer: ```suggestion
Crownlabs backend is written in Golang (a.k.a. Go), an open source, procedural programming language created by Google.
In brief, it looks similar to C and C++ but without some of their most tricky issues (e.g. memory management). This guide will do a general excursus on this fast raising programming language, introducing its most relevant features and giving a roadmap for a possible learning path.
```

## Useful links

### Install Go

The first step is to install Go and it can be done easly [here](https://golang.org/doc/install).

### Ides and editor plugins

> To write code in Go there are some ides like [Visual Studio Code](https://code.visualstudio.com/) or [Goland](https://www.jetbrains.com/go/) and text editor plugins like [vim-go](https://github.com/fatih/vim-go).

Reviewer: ```suggestion
To write code in Go, we suggest some IDEs such as [Visual Studio Code](https://code.visualstudio.com/) or [Goland](https://www.jetbrains.com/go/) and text editor plugins like [vim-go](https://github.com/fatih/vim-go).
```

For newbie students of Go the best is *Goland* which also has some mini tutorials of editing with code examples.

## Start learning Go

### Suggestions

Since Go is similar to other famous programming language (e.g. C/C++) it is very simple to learn sintax and for this purpose you can use [official documentation](https://golang.org/doc/) which is very simple and clear with a lot of tutorials and examples.

Very useful to the process of learning is [Tour of Go](https://tour.golang.org/list) where you can find many interactive tutorials to understand all parts of the language: you can do these little lectures in any order you like creating your personal learning path.

## Meet authors

> If you are interested in this lenguage you can better understand it and the idea behind it watching this [video](https://youtu.be/3yghHvvZQmA) where you can listen Go creators speaking at a Google Cloud conference. 

Reviewer: ```suggestion
If you are interested in this language, you can better understand it and the idea behind it watching this video [(Meet the Authors – Go Language)](https://youtu.be/3yghHvvZQmA) where you can listen Go creators speaking at a Google Cloud Next 2019 conference. 
```
